cpp/src/strings/combine.cu

```diff
@@ -50,21 +48,21 @@ std::unique_ptr<column> concatenate( table_view const& strings_columns,
     auto num_columns = strings_columns.num_columns();
     CUDF_EXPECTS( num_columns > 0, "At least one column must be specified");
     // check all columns are of type string
-    for( auto itr = strings_columns.begin(); itr != strings_columns.end(); ++itr )
-        CUDF_EXPECTS( itr->type().id()==STRING, "All columns must be of type string");
+    CUDF_EXPECTS( std::all_of( strings_columns.begin(), strings_columns.end(), [] (auto c) {return c.type().id()==STRING;}),
+                  "All columns must be of type string" );
     if( num_columns==1 ) // single strings column returns a copy
         return std::make_unique<column>(*(strings_columns.begin()),stream,mr);
     auto strings_count = strings_columns.num_rows();
     if( strings_count == 0 ) // empty begets empty
         return detail::make_empty_strings_column(mr,stream);
 
     CUDF_EXPECTS( separator.is_valid(), "Parameter separator must be a valid string_scalar");
-
-    auto execpol = rmm::exec_policy(stream);
     string_view d_separator(separator.data(),separator.size());
-    string_view d_narep(nullptr,0);
-    if( narep.is_valid() ) // narep is allowed to be invalid
-        d_narep = string_view(narep.data(),narep.size());
+    string_view const d_narep = [&] {
```

Comment:
Current guidelines is to name everything we capture.

Reply:
> Current guidelines is to name everything we capture.

What am I missing?

Reply:
I think he meant you have specify all the variables begin used rather than getting all.

Reply:
```suggestion
    string_view const d_narep = [&narep] {
```

Reply:
Ok, this is the IILE pattern I was taught by @codereport

Reply:
> I think he meant you have specify all the variables being used rather than getting all.

Yes, that's what I meant. https://github.com/rapidsai/cudf/pull/3597#discussion_r358960515

Reply:
Ah yes, very good.